extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.controller.js

```diff
@@ -114,22 +114,72 @@ define('ext.wikia.design-system.on-site-notifications.controller', [
 				}.bind(this));
 			},
 
-			markAsRead: function (uriAndType) {
+			markAsRead: function (notificationDetails) {
+				if (!notificationDetails.isUnread) {
+					return;
+				}
+
 				$.ajax({
 					type: 'POST',
-					data: JSON.stringify([uriAndType.uri]),
+					data: JSON.stringify([notificationDetails.uri]),
 					dataType: 'json',
-					contentType: "application/json; charset=UTF-8",
+					contentType: 'application/json; charset=UTF-8',
 					url: this.getBaseUrl() + '/notifications/mark-as-read/by-uri',
 					xhrFields: {
 						withCredentials: true
 					}
 				}).done(function () {
-					this._model.markAsRead(uriAndType.uri);
+					this._model.markAsRead(notificationDetails.uri);
 					this.updateUnreadCount();
 				}.bind(this));
 			},
 
+			markAsReadOnPageUnload: function (notificationDetails) {
+				if (!notificationDetails.isUnread) {
+					return;
+				}
+
+				var markAsReadUrl = this.getBaseUrl() + '/notifications/mark-as-read/by-uri';
+				var data = JSON.stringify([notificationDetails.uri]);
+
+				if (window.navigator.sendBeacon) {
+					try {
+						var blob = new Blob([data], {
+							type: 'application/json'
+						});
+
+						if (window.navigator.sendBeacon(markAsReadUrl, blob) === false) {
```

Comment:
If you'd check here for true (and return if it is true), you could use one `this.sendBeaconFallback` at the end